### PR TITLE
MAUI-2937-Appointment reminder in scheduler

### PR DESCRIPTION
**Pull request by Kerubananthan3761**
### Feature description ###
Provide  Reminder Alert  event to notify the reminder for scheduler appointments.

**Purpose/benefits of the feature**
This feature is used to remind the scheduler appointment before appointment's start time by using scheduler event ReminderAlertOpening event. 

**Use cases**
This feature is used to notify the scheduler reminder before the appointment start time and it can be use by enabling local notification or popup.

**What problem it solves?**

**Do competitors have this this feature? If possible, add comparison and also share relevant links. Anyhow, this should be available in requirements doc which you can link** 

https://syncfusion.sharepoint.com/:w:/r/sites/MAUI/_layouts/15/Doc.aspx?sourcedoc=%7BA4F3EE73-A7DF-4063-8B8B-3D389CB8F728%7D&file=MAUI-2266%20%E2%80%93%20Scheduler%20Reminder.docx&action=default&mobileredirect=true

### Analysis and design ###
https://syncfusion.sharepoint.com/:w:/r/sites/MAUI/_layouts/15/Doc.aspx?sourcedoc=%7BA4F3EE73-A7DF-4063-8B8B-3D389CB8F728%7D&file=MAUI-2266%20%E2%80%93%20Scheduler%20Reminder.docx&action=default&mobileredirect=true

### Solution description ###
- Added Dispatcher Timer To trigger Timer Tick event on Reminder Alert Time
- Added Scheduler Reminder class to get Appointment Reminder Properties
- Once an Appointment with reminder property is created, it has to be added Reminder Appointment Collection
- The Reminder Appointment with Alert Time will be Trigger ReminderAlertOpening Event on particular Time 
- The Reminder properties for both scheduler and custom appointments can be changed dynamically

Describe your code changes in detail for reviewers.

### Output screenshots ###

Post the output screenshots if an UI is affected or added due to this feature.

### Areas affected and ensured ###

List the areas are affected by your code changes.

### API Changes ###

https://syncfusion.atlassian.net/browse/MAUI-2652?oldIssueView=true 

https://syncfusion.sharepoint.com/:w:/r/sites/MAUI/_layouts/15/Doc.aspx?sourcedoc=%7B17DDEFC7-2E7F-4725-896C-6E00570631D6%7D&file=MAUI-2266_ReminderSupport.docx&action=default&mobileredirect=true&wdOrigin=TEAMS-ELECTRON.p2p.bim&wdExp=TEAMS-CONTROL&wdhostclicktime=1658224726500&cid=d9f74aed-f32a-43f2-8fa0-dbd824a8d37a

### Behavioral Changes ###
NA

### Test cases ###

Provide the unit testing written file details to understand the use cases considered in this implementation.
If there is no TDD (if it’s not possible to follow), provide the UI automation script location and the Excel file that contains the use cases considered in this implementation.
Provide the test cases Excel file alone if the feature cannot be automated in any case.

https://syncfusion.sharepoint.com/sites/Schedulemobileanddesktop/_layouts/OneNote.aspx?id=%2Fsites%2FSchedulemobileanddesktop%2FSiteAssets%2FSchedule%20mobile%20and%20desktop%20Notebook&wd=target%28MAUI.one%7C33201F6E-96D7-4940-8217-DED7488D6105%2FRecurrence%20reminder%7C45991912-8CC2-4008-ACC1-45A02725C022%2F%29

List out all the scenarios you have tested after include these changes. You can refer the predefined scenarios from below link.
- Property changes with Reminder time interval and IsDismissed for Scheduler reminder and custom object 
- Reminder collection changes need to test for Scheduler reminder and custom object 
- Recurrence exception appointment and Changed recurrence appointment. 
- Need to test with normal appointment, Span appointment, All day appointment, Recurrence and occurrence appointment 
- Schedule time zone and Appointment Time zone. 

### Testbed sample location ###

Provide the test bed sample location where code reviewers can review the new feature’s behaviors. This depends on the CI process that your team follows. It can be from NPMCI, HockeyApp, staging site, local server, etc.

### Does it have any known issues?

Due to framework restriction, the Reminder Popup is only achieved from Sample 

### MR CheckList ###

- [x] Have you ensured in iOS, Android, WinUI, and macOS(if supported)?
- [ ]  If there is any API change, did you get approval from PLO through [JIRA Tasks](https://syncfusion.atlassian.net/issues/?filter=66496)? **NA**
- [ ] Is there any existing behavior change of other features due to this code change? **NA**
- [ ] Have you enabled the necessary [settings](https://syncfusion.sharepoint.com/sites/MAUI/SitePages/Project-Settings.aspx) in your project, if you have created any new project? **NA**
- [x] Have you suppressed any warning or binding errors?
- [ ] Did you add a sample in the testbed for your feature? **NA**
- [ ] Did you record this case in the unit test or UI test? **NA**
- [x] Whether the new APIs and its comments are added as per [standard](https://syncfusion.sharepoint.com/sites/MAUI/SitePages/Coding.aspx)?
- [ ] Does it contain code that reflects any internal framework API? **NA**
- [ ] Have you included license for your control(If it is stable)? **NA**
- [ ] Did you ensure the cases mentioned in [this](https://syncfusion.sharepoint.com/sites/MAUI/SitePages/Testing-Scenarios.aspx) link? **NA**
- [x] Did you test the [memory](https://syncfusion.sharepoint.com/sites/MAUI/SitePages/Memory.aspx) leak with your feature?
- [x] Did you ensure the performance? Check [this](https://syncfusion.sharepoint.com/sites/MAUI/SitePages/Performance.aspx) link to know more about performance optimization and how to automate?
- [x] Does it need localization? If so, did you ensure the cases mentioned in [this](https://syncfusion.sharepoint.com/sites/MAUI/SitePages/Localization.aspx) link?
- [ ] Does it follow the [design system](https://syncfusion.sharepoint.com/sites/MAUI/SitePages/Design-System.aspx) guidelines and support light and dark themes? **NA**
- [ ] Did you ensure the new control / feature met [accessibility] (https://syncfusion.sharepoint.com/sites/MAUI/SitePages/Accessibility-in-.NET-MAUI.aspx) requirements? **NA**
- [ ] Did you ensure [RTL](https://syncfusion.sharepoint.com/sites/MAUI/SitePages/RTL%20(Flow%20Direction).aspx)? **NA**
- [ ] If you added any interaction related code, have you used touch and gesture APIs from core project? **NA**
- [ ] If you use a third-party package, did you get approval to use it? If not, please get approval before merging. **NA**

### CODE REVIEW

File: MAUI/Scheduler/reminders.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    var reminders = e.Reminders;
+    e.Reminders[0].IsDismissed = true;
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Handling reminders for future appointment
+
+* If the Reminder alert time is less than the current time, then appointment reminder time will be calculated from the current time.
+
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:216]: Add toc file for main title
https://github.com/syncfusion-content/maui-docs/blob/development/maui-toc.html

Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md:216]: Added toc file for main title

File: MAUI/Scheduler/reminders.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+## Handling reminders for overdue appointment
+
+* If a Non-recurring appointments reminder is in past, reminder alert as overdue 
+
+* If a recurring appointment is in the past such past occurrences do not have reminders. 
+
+* Span appointment < 24 hours - Shows alert time in overdue from appointment 
+
+* Overdue after recurrence end 
+
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:228]: ## Handling reminders for future appointment

•	Future normal and recurring appointment reminder alert will be displayed as Reminder alert time. 
•	Reminder alert time will be calculated form appointment start time substrate with reminder time interval. For an example if appointment time is 3.30pm and reminder time interval is new Timespan (0,15,0) then reminder alter time is 3.15 pm. 
•	If the Reminder alert time is less than the current time, then appointment reminder time will be calculated from the current time. 


## Handling reminders for overdue appointment

•	If a Non-recurring appointments reminder is in past, reminder alert as overdue until dismissed.
•	Recurring appointment overdue after recurrence end until dismissed and past occurrences do not have reminders.
•	Recurrence appointment due calculated from current date occurrence and its in due until next occurrence reminder time. 
•	If the last occurrence in recurrence is series is get dismissed, then the pattern recurring appointment dismissed. 

## Handling reminder for Changed occurrence appointment

•	Once the changed occurrence is moved to the future, its reminder is re-created from the pattern.
•	When the changed occurrence is restored to a time in the past, the reminder is deleted. 
•	Past Changed occurrence reminders will be dismissed.

Handle Dismissed for reminder
•	Normal appointment directly dismissed using 
•	For recurrence appointment, if current occurrence need to dismiss then need to add changed occurrence for reminder occurrence dismissed, then current occurrence dismissed details get updated in underlying appointment or data source. 
•	If only occurrence dismissed, then the changed icon will not be updated for dismissed changed occurrence 
Handle Snooze for reminder

File: MAUI/Scheduler/reminders.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+title: Reminders in .NET MAUI Scheduler control | Syncfusion
+description: Learn here all about the Reminder support in Syncfusion .NET MAUI Scheduler (SfScheduler) control, its element and more.
+platform: maui
+control: SfScheduler
+documentation: ug
+---
+
+# Reminder in .NET MAUI Scheduler (SfScheduler)
+
+The MAUI Scheduler notify an appointment reminder by using  the [ReminderAlertOpening] event. An appointment can have one or more reminders.
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:12]: Add restriction about restriction

File: MAUI/Scheduler/reminders.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+{% endtabs %}
+
+## ReminderAlertOpening event
+Scheduler notify the appointment's reminder by [ReminderAlertOpening] event before the appointment's start time. The [ReminderAlertOpeningEventArgs] has the following property,
+* [Reminders]: Gets a list of reminders that are used to notify the appointment reminders.
+
+{% tabs %}
+{% highlight c# tabtitle="MainWindow.xaml.cs" hl_lines="1" %}
+scheduler.ReminderAlertOpening += Scheduler_ReminderAlertOpening;
+
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:204]: Handle Dismissed for reminder
•	Normal appointment directly dismissed using 

•	For recurrence appointment, if current occurrence need to dismiss then need to add changed occurrence for reminder occurrence dismissed, then current occurrence dismissed details get updated in underlying appointment or data source. 
•	If only occurrence dismissed, then the changed icon will not be updated for dismissed changed occurrence 
Handle Snooze for reminder

File: MAUI/Scheduler/reminders.md (hunk 3 above)
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:12]: The MAUI Scheduler notify an appointment reminder by using the EnableReminder property and [ReminderAlertOpening] event

File: MAUI/Scheduler/reminders.md (hunk 2 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md:236]: Changes updated.

File: MAUI/Scheduler/reminders.md (hunk 3 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Added Note about restriction.

File: MAUI/Scheduler/reminders.md (hunk 4 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md:210]: Updated the content.

File: MAUI/Scheduler/reminders.md (hunk 3 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Changes updated.

File: MAUI/Scheduler/reminders.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+Map those properties of the `Event` class with the [SfScheduler] control by using the [AppointmentMapping] and map `CustomReminder` properties with the [SchedulerReminder] by using [ReminderMapping.]
+
+{% tabs %}
+{% highlight xaml tabtitle="MainWindow.xaml" hl_lines="3 17 18 19 20" %}
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:155]: Update proper syntex
MainWindow.xaml

File: MAUI/Scheduler/reminders.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+{% endtabs %}
+## Handle Dismissed for reminder
+
+* Normal appointment directly dismissed using [IsDismissed] property
+
+* For recurrence appointment, if current occurrence need to dismiss then need to add changed occurrence for reminder occurrence dismissed, then current occurrence dismissed details get updated in underlying appointment or data source.
+
+* If only occurrence dismissed, then the changed icon will not be updated for dismissed changed occurrence
+
+## Handle Snooze for reminder
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:226]: Add example code for snooze and Dismissed

File: MAUI/Scheduler/reminders.md (hunk 5 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Updated with proper syntex.

File: MAUI/Scheduler/reminders.md (hunk 6 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md:375]: Added example code for snooze and Dismissed

File: MAUI/Scheduler/reminders.md [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+                    StartTime = new DateTime(changedExceptionDate.Year, changedExceptionDate.Month, changedExceptionDate.Day, changedExceptionDate.Hour, 0, 0),
+                    EndTime = new DateTime(endDate.Year, endDate.Month, endDate.Day, endDate.Hour, 0, 0),
+                    Background = appointment.Background,
+                    RecurrenceId = 1,
+                    Reminders = Reminder
+                };
+                if (!(scheduler.AppointmentsSource as ObservableCollection<SchedulerAppointment>).Contains(exceptionAppointment))
+                {
+                    (scheduler.AppointmentsSource as ObservableCollection<SchedulerAppointment>).Add(exceptionAppointment);
+                    exceptionAppointment.Reminders[0].IsDismissed = true;
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:430]: Remove - 
 exceptionAppointment.Reminders[0].IsDismissed = true;

File: MAUI/Scheduler/reminders.md [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+                // Clone parent details
+                var exceptionAppointment = new SchedulerAppointment()
+                {
+                    Id = 2,
+                    Subject = appointment.Subject,
+                    StartTime = new DateTime(changedExceptionDate.Year, changedExceptionDate.Month, changedExceptionDate.Day, changedExceptionDate.Hour, 0, 0),
+                    EndTime = new DateTime(endDate.Year, endDate.Month, endDate.Day, endDate.Hour, 0, 0),
+                    Background = appointment.Background,
+                    RecurrenceId = 1,
+                    Reminders = Reminder
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:425]: Remove - 
 var Reminder = new ObservableCollection<SchedulerReminder>()
            {
                new SchedulerReminder {TimeBeforeStart = new TimeSpan (0,0,30)}
            };

Use directly
new SchedulerReminder {TimeBeforeStart = new TimeSpan (0,0,30)}

File: MAUI/Scheduler/reminders.md [hunk 9]
```diff
@@ -0,0 +1,444 @@
...
+        var appointment = e.Reminders[0].Appointment;
+        var Reminder = new ObservableCollection<SchedulerReminder>()
+        {
+            new SchedulerReminder {TimeBeforeStart = new TimeSpan (0,0,30)}
+        };
+        DateTime changedExceptionDate = e.Reminders[0].Appointment.StartTime;
+        DateTime endDate = e.Reminders[0].Appointment.EndTime;
+        if (!string.IsNullOrEmpty(e.Reminders[0].Appointment.RecurrenceRule))
+        {
+            appointment.RecurrenceExceptionDates = new ObservableCollection<DateTime>()
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:329]: patternAppointment = Schedule.AppointmentSource(x=> x.id == appointment.id)

patternAppointment.ExceptionDates

File: MAUI/Scheduler/reminders.md [hunk 10]
```diff
@@ -0,0 +1,425 @@
...
+            {
+                Id = 2,
+                Subject = patternAppointment.Subject,
+                StartTime = new DateTime(changedExceptionDate.Year, changedExceptionDate.Month, changedExceptionDate.Day, changedExceptionDate.Hour, 0, 0),
+                EndTime = new DateTime(endDate.Year, endDate.Month, endDate.Day, endDate.Hour, 0, 0),
+                Background = patternAppointment.Background,
+                RecurrenceId = 1,
+                Reminders = patternAppointment.Reminders,
+            };
+            if (!(scheduler.AppointmentsSource as ObservableCollection<SchedulerAppointment>).Contains(exceptionAppointment))
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:337]: Avoid repeated casting

File: maui-toc.html [hunk 11]
```diff
@@ -378,10 +378,11 @@
             <li><a href="/maui/Scheduler/timeline-views">Timeline Views</a></li>
             <li><a href="/maui/Scheduler/month-view">Month View</a></li>
             <li><a href="/maui/Scheduler/agenda-view">Agenda View</a></li>
-			<li><a href="/maui/Scheduler/resource-view">Resource View</a></li>
+            <li><a href="/maui/Scheduler/resource-view">Resource View</a></li>
             <li><a href="/maui/Scheduler/appointments">Appointments</a></li>
-			<li><a href="/maui/Scheduler/calendar-types">Calendar Types</a></li>
+            <li><a href="/maui/Scheduler/calendar-types">Calendar Types</a></li>
             <li><a href="/maui/Scheduler/load-on-demand">Load On Demand</a></li>
+            <li><a href="/maui/Scheduler/reminders">Reminder</a></li>
```
Comment (Jeyasri-Murugan) [inline maui-toc.html:385]: <li><a href="/maui/Scheduler/reminders">Reminders</a></li>

File: MAUI/Scheduler/reminders.md [hunk 12]
```diff
@@ -0,0 +1,425 @@
+---
+layout: post
+title: Appointment Reminders in .NET MAUI Scheduler control | Syncfusion
+description: Learn here all about the Reminder support in Syncfusion .NET MAUI Scheduler (SfScheduler) appointments.
+platform: maui
+control: SfScheduler
+documentation: ug
+---
+
+# Appointment reminder in .NET MAUI Scheduler (SfScheduler)
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:10]: Appointments reminders in .NET MAUI Scheduler (SfScheduler)

File: MAUI/Scheduler/reminders.md [hunk 13]
```diff
@@ -0,0 +1,425 @@
...
+
+* For All day appointment, TimeBeforeStart = Appointment.StartTime.Date.AddSeconds(DateTime.Now.Second) - DateTime.Now - snoozeTime
+
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="1" %}
+scheduler.ReminderAlertOpening += Scheduler_ReminderAlertOpening;
+
+private void Scheduler_ReminderAlertOpening(object sender, ReminderAlertOpeningEventArgs e)
+{
+    var reminders = e.Reminders;
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:372]: Remmove unwanted codes - var reminders = e.Reminders;

File: MAUI/Scheduler/reminders.md [hunk 14]
```diff
@@ -0,0 +1,425 @@
...
+                // Clone parent details
+                SchedulerAppointment exceptionAppointment = new SchedulerAppointment()
+                {
+                    Id = 2,
+                    Subject = patternAppointment.Subject,
+                    StartTime = new DateTime(changedExceptionDate.Year, changedExceptionDate.Month, changedExceptionDate.Day, changedExceptionDate.Hour, 0, 0),
+                    EndTime = new DateTime(endDate.Year, endDate.Month, endDate.Day, endDate.Hour, 0, 0),
+                    Background = patternAppointment.Background,
+                    RecurrenceId = 1,
+                    Reminders = patternAppointment.Reminders,
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/reminders.md:414]: Reminders = e.Reminders;

File: MAUI/Scheduler/reminders.md (hunk 7 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Removed exceptionAppointment.Reminders[0].IsDismissed = true;

File: MAUI/Scheduler/reminders.md (hunk 8 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Removed codes

File: MAUI/Scheduler/reminders.md (hunk 9 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Codes Changed

File: MAUI/Scheduler/reminders.md (hunk 12 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Changed to Appointments reminders in .NET MAUI Scheduler (SfScheduler)

File: MAUI/Scheduler/reminders.md (hunk 14 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Code changed

File: MAUI/Scheduler/reminders.md [hunk 15]
```diff
@@ -0,0 +1,423 @@
...
+title: Appointments Reminders in .NET MAUI Scheduler control | Syncfusion
+description: Learn here all about the appointments reminders support in Syncfusion .NET MAUI Scheduler (SfScheduler) appointments.
+platform: maui
+control: SfScheduler
+documentation: ug
+---
+
+# Appointments reminders in .NET MAUI Scheduler (SfScheduler)
+
+The MAUI Scheduler notify an appointment reminders by using  the [EnableReminder] property and [ReminderAlertOpening] event. An appointment can have one or more reminders.
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:12]: MAUI  -> use the Term ".Net MAUI" - correct elsewhere

File: MAUI/Scheduler/reminders.md [hunk 16]
```diff
@@ -0,0 +1,423 @@
...
+
+N>
+* As restriction in enabling toast notification in .NET MAUI Framework, as of now added event to notify appointment reminder.[Reference](https://github.com/dotnet/maui/discussions/4216)
+
+## Handling reminders for future appointment
+
+* Future normal and recurring appointment reminder alert will be displayed as Reminder alert time.
+
+* Reminder alert time will be calculated form appointment start time substrate with reminder time interval. For an example if appointment time is 3.30pm and reminder time interval is new Timespan (0,15,0) then reminder alter time is 3.15 pm.
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:21]: new Timespan (0,15,0) - add this as technical term

File: MAUI/Scheduler/reminders.md [hunk 17]
```diff
@@ -0,0 +1,423 @@
...
+
+## Handling reminders for overdue appointment
+
+* If a Non-recurring appointments reminder is in past, reminder alert as overdue untill dismissed.
+
+* Recurring appointment overdue after recurrence end until dismissed and past occurrences do not have reminders.
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:27]: and past occurrences do not have reminders. - what are you conveying here?

File: MAUI/Scheduler/reminders.md [hunk 18]
```diff
@@ -0,0 +1,423 @@
...
+
+* Recurrence appointment due calculated from current date occurrence and its in due until next occurrence reminder time.
+
+*  If the last occurrence in recurrence is series is get dismissed, then the pattern recurring appointment dismissed.
+
+## Handling reminders for Changed occurrence appointment
+
+* Once the changed occurrence is moved to the future, its reminder is re-created from the pattern.
+
+* When the changed occurrence is restored to a time in the past, the reminder is deleted.
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:37]: the reminder is deleted - ?

File: MAUI/Scheduler/reminders.md [hunk 19]
```diff
@@ -0,0 +1,423 @@
...
+
+* Recurring appointment overdue after recurrence end until dismissed and past occurrences do not have reminders.
+
+* Recurrence appointment due calculated from current date occurrence and its in due until next occurrence reminder time.
+
+*  If the last occurrence in recurrence is series is get dismissed, then the pattern recurring appointment dismissed.
+
+## Handling reminders for Changed occurrence appointment
+
+* Once the changed occurrence is moved to the future, its reminder is re-created from the pattern.
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:35]: Explain this  its reminder is re-created from the pattern.

File: MAUI/Scheduler/reminders.md [hunk 20]
```diff
@@ -0,0 +1,423 @@
...
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="1" %}
+scheduler.ReminderAlertOpening += Scheduler_ReminderAlertOpening;
+
+private void Scheduler_ReminderAlertOpening(object sender, ReminderAlertOpeningEventArgs e)
+{
+    bool snooze = await DisplayAlert("Reminder", Reminders[0].Appointment.Subject + " - " + Reminders[0].Appointment.StartTime.ToString(" dddd, MMMM dd, yyyy, hh:mm tt"), "Snooze", "Dismiss"); 
+}
+{% endhighlight %}
+{% endtabs %}
+## Handle Dismissed for reminders
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:297]: Handle dismiss property of reminders

**Review by Nijamudeen-Mohamed — COMMENTED** [review]
Update my review comments

File: MAUI/Scheduler/reminders.md [hunk 21]
```diff
@@ -0,0 +1,423 @@
+---
+layout: post
+title: Appointments Reminders in .NET MAUI Scheduler control | Syncfusion
+description: Learn here all about the appointments reminders support in Syncfusion .NET MAUI Scheduler (SfScheduler) appointments.
+platform: maui
+control: SfScheduler
+documentation: ug
+---
+
+# Appointments reminders in .NET MAUI Scheduler (SfScheduler)
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:10]: Appointment reminders

File: MAUI/Scheduler/reminders.md [hunk 22]
```diff
@@ -0,0 +1,423 @@
...
+
+The MAUI Scheduler notify an appointment reminders by using  the [EnableReminder] property and [ReminderAlertOpening] event. An appointment can have one or more reminders.
+
+N>
+* As restriction in enabling toast notification in .NET MAUI Framework, as of now added event to notify appointment reminder.[Reference](https://github.com/dotnet/maui/discussions/4216)
+
+## Handling reminders for future appointment
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:17]: Handling reminders for future appointments

File: MAUI/Scheduler/reminders.md [hunk 23]
```diff
@@ -0,0 +1,423 @@
...
+
+* Future normal and recurring appointment reminder alert will be displayed as Reminder alert time.
+
+* Reminder alert time will be calculated form appointment start time substrate with reminder time interval. For an example if appointment time is 3.30pm and reminder time interval is new Timespan (0,15,0) then reminder alter time is 3.15 pm.
+
+## Handling reminders for overdue appointment
```
Comment (Nijamudeen-Mohamed) [inline MAUI/Scheduler/reminders.md:23]: overdue appointments

File: MAUI/Scheduler/reminders.md [hunk 24]
```diff
@@ -0,0 +1,423 @@
...
+
+* Recurring appointment overdue from last occurrence date and time until last occurrence is dismissed.
+
+* Recurrence appointment due calculated from current date occurrence and it is in due until next occurrence reminder time.
+
+*  If the last occurrence in recurrence is series is get dismissed, then the pattern recurring appointment dismissed.
+
+## Handling reminders for Changed occurrence appointments
+
+* Once the changed occurrence is moved to the future, then pattern reminder value will be used for changed occurrence.
```
Comment (florenceliliana) [inline MAUI/Scheduler/reminders.md:35]: * Once the changed occurrence is moved to the future, then the pattern reminder value will be used for the changed occurrence.

File: MAUI/Scheduler/reminders.md [hunk 25]
```diff
@@ -0,0 +1,423 @@
...
+
+* When the changed occurrence is restored to a time in the past, then its reminder will be dismissed.
+
+* Past Changed occurrence reminders will be dismissed.
+
+## Enable reminders
+
+Reminders can be enabled by setting the [EnableReminder] property to `true` which will trigger the `ReminderAlertOpening` event to notify appointment reminders. The reminders can be set by using the [Reminders] property of [SchedulerAppointment.]
```
Comment (florenceliliana) [inline MAUI/Scheduler/reminders.md:43]: Reminders can be enabled by setting the [EnableReminder] property to `true,` which will trigger the `ReminderAlertOpening` event to notify appointment reminders. The reminders can be set by using the [Reminders] property of [SchedulerAppointment].

File: MAUI/Scheduler/reminders.md [hunk 26]
```diff
@@ -0,0 +1,423 @@
...
+        public string RecurrenceRule { get; set; }
+        public ObservableCollection<DateTime> RecurrenceExceptions { get; set; }
+        public ObservableCollection<Reminder> Reminders { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+The [ReminderMapping] provides the mapping information about the [SchedulerReminder] properties to the [DataItem] object. `ReminderMapping` has the following properties.
+
+* [TimeBeforeStart]: Maps the property name of a custom class, which is equivalent for the [SchedulerReminder.TimeBeforeStart.]
```
Comment (florenceliliana) [inline MAUI/Scheduler/reminders.md:169]: * [TimeBeforeStart]: Maps the property name of a custom class, which is equivalent to the [SchedulerReminder.TimeBeforeStart].

File: MAUI/Scheduler/reminders.md [hunk 27]
```diff
@@ -0,0 +1,423 @@
...
+
+* [IsDismissed]: Maps the property name of a custom class, which is equivalent for the [SchedulerReminder.IsDismissed.]
```
Comment (florenceliliana) [inline MAUI/Scheduler/reminders.md:171]: * [IsDismissed]: Maps the property name of a custom class, which is equivalent to the [SchedulerReminder.IsDismissed].

File: MAUI/Scheduler/reminders.md [hunk 28]
```diff
@@ -0,0 +1,423 @@
...
+    /// Gets or sets the value indicating whether the reminder is dismissed or not. 
+    /// </summary>
+    public bool IsDismissed { get; set; }
+
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+Map those properties of the `Event` class with the [SfScheduler] control by using the [AppointmentMapping] and map `CustomReminder` properties with the [SchedulerReminder] by using [ReminderMapping.]
```
Comment (florenceliliana) [inline MAUI/Scheduler/reminders.md:195]: Map those properties of the `Event` class with the [SfScheduler] control by using the [AppointmentMapping] and map `CustomReminder` properties with the [SchedulerReminder] by using [ReminderMapping].

File: MAUI/Scheduler/reminders.md [hunk 29]
```diff
@@ -0,0 +1,423 @@
...
+        }
+
+    };
+    Events.Add(recurrenceAppointment);
+}
+{% endhighlight %}
+{% endtabs %}
+
+## ReminderAlertOpening event
+Scheduler notify the appointment's reminders by [ReminderAlertOpening] event before the appointment's start time. The [ReminderAlertOpeningEventArgs] has the following property,
```
Comment (florenceliliana) [inline MAUI/Scheduler/reminders.md:284]: Scheduler notifies the appointment's reminders by [ReminderAlertOpening] event before the appointment's start time. The [ReminderAlertOpeningEventArgs] has the following property.

**Review by florenceliliana — CHANGES_REQUESTED** [review]

File: MAUI/Scheduler/reminders.md (hunk 15 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Updated as ".Net MAUI".

File: MAUI/Scheduler/reminders.md (hunk 16 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Added new Timespan (0,15,0) as technical term

File: MAUI/Scheduler/reminders.md (hunk 17 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Changed as - Recurring appointment overdue from the last occurrence date and time until the last occurrence is dismissed.

File: MAUI/Scheduler/reminders.md (hunk 18 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Changed as dismissed

File: MAUI/Scheduler/reminders.md (hunk 19 above)
Comment (Kerubananthan3761) [reply MAUI/Scheduler/reminders.md]: Changed as - Once the changed occurrence is moved to the future, then the pattern reminder value will be used for the changed occurrence.

**Review by Jeyasri-Murugan — APPROVED** [review]

**Pull request merged**